stdlib/Printf/src/Printf.jl

```diff
@@ -409,11 +411,13 @@ const __BIG_FLOAT_MAX__ = 8192
                 if len > siz
                     maxout = max(__BIG_FLOAT_MAX__,
                                  ceil(Int, precision(x) * log(2) / log(10)) + 25)
+                    # TODO: `error` is kind of generic, is there a more appropriate one to throw?
                     len > maxout &&
                         error("Over $maxout bytes $len needed to output BigFloat $x")
```

Comment:
I think getting a better error message here requires knowing how to get to this code path in the first place? This hinges on knowing what `_snprintf` returns, which seems cryptic and not corresponding to the variable name `len` implies at all:

> Return the number of characters that would have been written had n been sufficiently large, not counting the terminating null character, or a negative value if an error occurred.

Ref https://www.mpfr.org/mpfr-current/mpfr.html, search for `mpfr_snprintf` if I'm not mistaken.

Where does `buf` come from, why does `__BIG_FLOAT_MAX__` exist (does this mean we can only print `BigFloat` whose representation takes up less than 8KB)?